amazonfreertossdk/src/main/java/software/amazon/freertos/amazonfreertossdk/AmazonFreeRTOSDevice.java

```diff
@@ -536,6 +571,7 @@ public void onCharacteristicRead(BluetoothGatt gatt,
                                 Log.i(TAG, "Default MTU is set to: " + currentMtu.mtu);
                                 try {
                                     mMtu = Integer.parseInt(currentMtu.mtu);
+                                    mMaxPayloadLen = mMtu - 3;
                                     if (mDeviceInfoCallback != null) {
```

Comment:
setMtu() is called only if mDeviceInfoCallback is not null, hence if we move the setting to BLE_INITIALIZED state into onMtuChange, it might never be set.